### PR TITLE
Enable the graph mode and add the full warmup logic for Deepseek OCR model

### PR DESCRIPTION
**Pull request by HeJunyan**
The Deepseek OCR model uses fixed input tensor shape, so there is no need for padding.
Its input is different from the other MM models so a new fake data set is added to warm it up.

### CODE REVIEW

File: vllm/worker/hpu_model_runner.py [hunk 1]
```diff
@@ -107,14 +107,18 @@ class VisionBuckets:
     This class is used to bucket image tokens
     '''
 
-    def __init__(self, is_batch_based):
+    def __init__(self, is_batch_based, sub_image_list=None):
```
Comment (czhu15) [inline vllm/worker/hpu_model_runner.py]: suggest to add the description of the argument of is_batch_based and sub_image_list in the defination.
sub_image_list seems like a list of image objects. But from line 117, maybe it is a list of integers?

File: vllm/worker/hpu_model_runner.py [hunk 2]
```diff
@@ -206,6 +210,11 @@ def is_mm_optimized(model):
         'DeepseekOCRForCausalLM' in str(type(model))
 
 
+def fixed_sub_image_list(model):
+    return [i for i in range(6+1) if i!=1] \
```
Comment (czhu15) [inline vllm/worker/hpu_model_runner.py:219]: this code is hard to understand. why just not [0,2,3,4,5,6] and explain how this list comes?

File: vllm/worker/hpu_model_runner.py [hunk 3]
```diff
@@ -206,6 +210,11 @@ def is_mm_optimized(model):
...
+        if 'DeepseekOCRForCausalLM' in str(type(model)) else None
```
Comment (czhu15) [inline vllm/worker/hpu_model_runner.py]: the model detection via str(type(model)) is brittle—prefer isinstance if possible.

File: vllm/worker/hpu_model_runner.py [hunk 4]
```diff
@@ -408,9 +417,11 @@ def __init__(self, model, vllm_config, is_causal, sampler):
...
         # the static and dynamic parts distinct.
         if htorch.utils.internal.is_lazy():
-            if self.model_is_mrope and hasattr(self.model, 'visual') and \
-               model_config is not None and \
-               model_config.model_type not in ("glm4v_moe", "paddleocr_vl"):
+            if (self.model_is_mrope and hasattr(self.model, 'visual') and
+                model_config is not None
+                and model_config.model_type not in
+                ("glm4v_moe", "paddleocr_vl")) \
+               or 'DeepseekOCRForCausalLM' in str(type(self.model)):
```
Comment (czhu15) [inline vllm/worker/hpu_model_runner.py]: the model detection via str(type(model)) is brittle—prefer isinstance if possible.

File: vllm/worker/hpu_model_runner.py (hunk 1 above)
Comment (HeJunyan) [reply vllm/worker/hpu_model_runner.py]: Done

File: vllm/worker/hpu_model_runner.py (hunk 2 above)
Comment (HeJunyan) [reply vllm/worker/hpu_model_runner.py:219]: Done

File: vllm/worker/hpu_model_runner.py (hunk 3 above)
Comment (HeJunyan) [reply vllm/worker/hpu_model_runner.py]: OK

File: vllm/worker/hpu_model_runner.py (hunk 4 above)
Comment (HeJunyan) [reply vllm/worker/hpu_model_runner.py]: OK

**Comment by HeJunyan** [comment]
I verified the PR in full warmup mode, the result is unchanged. So it is OK.

**Review by czhu15 — APPROVED** [review]
LGTM
